server/src/main/java/org/apache/iotdb/db/qp/executor/QueryProcessExecutor.java

```diff
@@ -244,18 +271,47 @@ public boolean insert(InsertPlan insertPlan) throws ProcessorException {
     try {
       String[] measurementList = batchInsertPlan.getMeasurements();
       String deviceId = batchInsertPlan.getDeviceId();
-      MNode node = mManager.getNodeByDeviceIdFromCache(deviceId);
+      Object[] values = collectFirstElements(batchInsertPlan.getColumns());
```

Comment:
I think you should delay the collection because not all insertions need schema creation.